importer/src/path_canonicalizer.rs

```diff
@@ -571,10 +579,17 @@ fn check_machine(
             return Err(format!("Duplicate name `{name}` in machine `{location}`"));
         }
     }
+    for param in &m.params.0 {
```

Comment:
I fear we have to do that everywhere where Param appears...

Reply:
what do you mean?

Reply:
You changed the parser to parse parameter types as paths instead of identifiers. Now you have to resolve all of them to absolute paths in the path canonicalizer, not just for the machine arguments.

Reply:
this `Param` type is quite overloaded already... but in other usages `ty` must be an identifier (or must be None).
I changed it to `SymbolPath` because it still allows for parsing identifiers (using `try_to_identifier` to get the value later).
I don't see why those cases which only allow for identifiers need to be handled by the path canonicalizer (they should not be paths).
Maybe I should not be using `Param` here?

Reply:
I mean my point is if you use SymbolPath and we don't resolve it, it will create problems later on.

Reply:
Ok, maybe it's fine for now. We will hopefully fix that with the rewrite.